backend/app.py

```diff
@@ -183,64 +172,121 @@ def delete_node(hierarchy_id, node_id):
-    #             "measure": 24,
-    #             "local_value": 64,
-    #             "global_value": 128,
-    #         }
-    #     ]
-    # }
-    
+    # Get all measurement nodes of the given hierarchy
+    measurements = Node.query.filter(Node.measurement_type != None, Node.hierarchy_id == hierarchy_id)
+    validNodeIds = []
+    for measurement in measurements:
+        validNodeIds.append(measurement.id)
+
+    data = request.get_json()
+
     # Create alternative
-    # Generate values with alternative_id
-        # Values must match up with measurement nodes
-        # Generate null values if a value is not provided for a measurement
+    alternative = Alternative(
+        hierarchy_id,
+        name=data["name"]
+    )
 
-    pass
+    if "values" in data:
+        for value in data["values"]:
+
+            # Generate null values if a value is not provided for a measurement
+            if "measure" in value:
+                measure = value["measure"]
+            else:
+                measure = None
+
+            if "local_value" in value:
+                local_value = value["local_value"]
+            else:
+                local_value = None
+                
+            if "global_value" in value:
+                global_value = value["global_value"]
+            else:
+                global_value = None
+
+            # Generate values with alternative_id
+            # Values must match up with measurement nodes
+            if value["nodeId"] in validNodeIds:
+                value = Value(value["nodeId"], measure, local_value, global_value)
+                alternative.values.append(value)
+            else:
+                abort(404, description="Node ID not valid")
+
+    db.session.add(alternative)
+    db.session.commit()
+
+    return alternative.to_dict(), 201
 
 
 # TODO: Get Alternative
 @app.route("/hierarchy/<hierarchy_id>/alternative/<alternative_id>")
-def get_alternative(hierarchy_id, alternative_id):
+def get_alternative(hierarchy_id, alternative_id, methods=['GET']):
     hierarchy = Hierarchy.query.filter_by(id=hierarchy_id).first()
-    alternative = Node.query.filter_by(id=alternative_id).first()
+    if not hierarchy:
+        abort(404, description="Resource not found")
+
+    alternative = Alternative.query.filter_by(id=alternative_id).first()
+    if not alternative:
+        abort(404, description="Resource not found")
 
-    # Return alternative.to_dict()
+    return alternative.to_dict(), 200
 
 
 # TODO: Delete Alternative
-@app.route("/hierarchy/<hierarchy_id>/alternative/<alternative_id>")
+@app.route("/hierarchy/<hierarchy_id>/alternative/<alternative_id>", methods=['DELETE'])
 def delete_alternative(hierarchy_id, alternative_id):
-    pass
+    hierarchy = Hierarchy.query.filter_by(id=hierarchy_id).first()
+    if not hierarchy:
+        abort(404, description="Resource not found")
+
+    alternative = Alternative.query.filter_by(id=alternative_id).first()
+    if not alternative:
```

Comment:
Same comment as above. I don't think it's necessary to query hierarchy separately unless you need data specific to a hierarchy.